core/src/main/scala/io/delta/tables/DeltaOptimizeBuilder.scala

```diff
@@ -39,7 +39,7 @@ class DeltaOptimizeBuilder private(
     sparkSession: SparkSession,
     tableIdentifier: String,
     options: Map[String, String]) extends AnalysisHelper {
-  @volatile private var partitionFilter: Option[String] = None
+  @volatile private var partitionFilter: Seq[Option[String]] = Seq.empty
```

Comment:
Just found we may be able to write a cleaner code if we change the type to `Seq[String]`? Sorry. I should have found this earlier.

Could you also remove `@volatile` since you are touching the code? It's not needed.

Reply:
removed volatile . if you don't mind me asking what is the use Annotation volatile here ?